pkg/dss/auth/scopes.go

```diff
@@ -0,0 +1,34 @@
+package auth
+
+// Operation models the name of an operation.
+type Operation string
+
+// String returns the string representation of o.
+func (o Operation) String() string {
+	return string(o)
+}
+
+// Scope models an oauth scope.
+type Scope string
+
+// String returns the string representation of s.
+func (s Scope) String() string {
+	return string(s)
+}
+
+// MergeOperationsAndScopes merges a and be together.
+func MergeOperationsAndScopes(requiredScopes ...map[Operation][]Scope) map[Operation][]Scope {
+	if len(requiredScopes) == 0 {
+		return nil
```

Comment:
nit: Why not just return an empty map rather than nil?

Reply:
Semantics are the same mostly, except for inserting into a `nil` map. I don't have a strong opinion here, will adjust as you proposed.